be/src/service/mem_hook.cpp

```diff
@@ -219,6 +219,7 @@ void operator delete[](void* p, size_t size, std::align_val_t al) noexcept {
 #define TRY_MEM_CONSUME(size, err_ret)                                                                   \
     do {                                                                                                 \
         if (LIKELY(starrocks::tls_is_thread_status_init)) {                                              \
+            RETURN_IF_UNLIKELY(!starrocks::tls_thread_status.try_singleton_check(size), err_ret);        \
```

Comment:
Not check for one batch(2m), may be have performance problem

Reply:
Add batch.